modules/data.remote/R/GEDI_AGB_prep.R

```diff
@@ -0,0 +1,527 @@
+#' coordinates. The footprints are located within the global latitude band
+#' observed by the International Space Station (ISS), nominally 51.6 degrees N
+#' and S and reported for the period 2019-04-18 to 2020-09-02
+#'
+#' @param ul_lat Numeric: upper left latitude.
+#' @param lr_lat Numeric: lower right latitude.
+#' @param ul_lon Numeric: upper left longitude.
+#' @param lr_lon Numeric: lower right longitude.
+#' @param ncore Numeric: numbers of core to be used if the maximum core
+#'   available is less than the number of files to be download. Default to the
+#'   number of cores available minus one.
+#' @param from Character: date from which the data search starts. In the form
+#'   "yyyy-mm-dd".
+#' @param to Character: date on which the data search end. In the form
+#'   "yyyy-mm-dd".
+#' @param outdir Character: path of the directory in which to save the
+#'   downloaded files.Default to the working directory. If it doesn't exist it
+#'   will be created. Ignored if \code{just_path=TRUE}
+#' @param just_path Logical: if TRUE return a character vector of available
+#'   files without downloading them. Default to FALSE.
+#' @param subset Numeric vector of indices for downloading a subset of files
+#'   instead of all. If is not numeric it will be ignored silently.
+#' @details During the first use, users will be ask to enter their Earth Explore
+#'   login Information for downloading the data. If you don't have already an
+#'   account, register
+#'   \href{https://urs.earthdata.nasa.gov/users/new?client_id=YQOhivHfMTau88rjbMOVyg&redirect_uri=https%3A%2F%2Fdaac.ornl.gov%2Fcgi-bin%2Furs%2Furs_logon_proc.pl&response_type=code&state=https%3A%2F%2Fdaac.ornl.gov%2Fcgi-bin%2Fdataset_lister.pl%3Fp%3D40}{at
+#'   this link}. These information will be saved in \code{outdir} as a netrc
+#'   file. This function uses the \code{\link[foreach]{foreach}} package for
+#'   downloading files in parallel, with the
+#'   \code{\link[doParallel]{doParallel}} configuration. If a file with the same
+#'   name is already presented in \code{outdir} it will be overwrite.
+#' @return List of file path in \code{outdir}
+#' @examples
+#'
+#'#retrive Italy bound
+#'bound <- sf::st_as_sf(raster::getData('GADM', country='ITA', level=1))
+#'ex <- raster::extent(bound)
+#'ul_lat <- ex[4]
+#'lr_lat <- ex[3]
+#'ul_lon <- ex[2]
+#'lr_lon <- ex[1]
+#'from <- "2020-07-01"
+#'to <- "2020-07-02"
+#'#get just files path available for the searched parameters
+#'l4_download(ul_lat=ul_lat,
+#'            lr_lat=lr_lat,
+#'            ul_lon=ul_lon,
+#'            lr_lon=lr_lon,
+#'            from=from,
+#'            to=to,
+#'            just_path=T
+#')
+#'
+#'#download the first 4 files
+#'
+#'l4_download(ul_lat=ul_lat,
+#'            lr_lat=lr_lat,
+#'            ul_lon=ul_lon,
+#'            lr_lon=lr_lon,
+#'            from=from,
+#'            to=to,
+#'            just_path=F,
+#'            outdir = tempdir(),
+#'            subset=1:4)
+#' @export
```

Comment:
repeat of previous question about which functions are internal vs exported.

Reply:
Fixed.